pkg/deploy/deploy.go

```diff
@@ -245,6 +245,16 @@ func (d *deployer) getParameters(ps map[string]interface{}) *arm.Parameters {
 
 	v := reflect.ValueOf(*d.config.Configuration)
 	for i := 0; i < v.NumField(); i++ {
+		if v.Field(i).IsNil() {
+			continue
+		}
+
+		val, ok := v.Field(i).Interface().(*string)
+		if ok {
+			m[strings.SplitN(v.Type().Field(i).Tag.Get("json"), ",", 2)[0]] = *val
```

Comment:
can you explain this for a bit?

Reply:
Idea is to convert the `*string -> string` here. It simplifies the code later on as the check does not have to be present anywhere else.

I have updated the code to be more descriptive. Thanks for pointing to the unreadable part.

Reply:
why not use `to.StringPtr()` ?